techlibs/quicklogic/synth_quicklogic.cc

```diff
@@ -0,0 +1,141 @@
+        log("\n");
+        log("    -edif <file>\n");
+        log("        write the design to the specified edif file. writing of an output file\n");
+        log("        is omitted if this parameter is not specified.\n");
+        log("\n");
+        log("    -blif <file>\n");
+        log("        write the design to the specified BLIF file. writing of an output file\n");
+        log("        is omitted if this parameter is not specified.\n");
+        log("\n");
+        help_script();
+        log("\n");
+    }
+
+    std::string top_opt = "";
+    std::string edif_file = "";
+    std::string blif_file = "";
+    std::string currmodule = "";
+
+    void clear_flags() override
+    {
+        top_opt = "-auto-top";
+        edif_file.clear();
+        blif_file.clear();
+    }
+
+    void execute(std::vector<std::string> args, RTLIL::Design *design) override
+    {
+        std::string run_from, run_to;
+        clear_flags();
+
+        size_t argidx;
+        for (argidx = 1; argidx < args.size(); argidx++)
+        {
+            if (args[argidx] == "-top" && argidx+1 < args.size()) {
+                top_opt = "-top " + args[++argidx];
+                continue;
+            }
+            if (args[argidx] == "-edif" && argidx+1 < args.size()) {
+                edif_file = args[++argidx];
+                continue;
+            }
+            if (args[argidx] == "-blif" && argidx+1 < args.size()) {
+                blif_file = args[++argidx];
+                continue;
+            }
+            break;
+        }
+        extra_args(args, argidx, design);
+
+        if (!design->full_selection())
+            log_cmd_error("This command only operates on fully selected designs!\n");
+
+        log_header(design, "Executing SYNTH_QUICKLOGIC pass.\n");
+        log_push();
+
+        run_script(design, run_from, run_to);
+
+        log_pop();
+    }
+
+    void script() override
+    {
+        if (check_label("begin")) {
+            run("read_verilog -lib +/quicklogic/cells_sim.v");
+            run(stringf("hierarchy -check %s", top_opt.c_str()));
+        }
+
+        if (check_label("prepare")) {
+            run("proc");
+            run("flatten");
+            run("wreduce -keepdc");
+            run("muxpack");
+        }
+
+        if (check_label("coarse")) {
+            run("tribuf -logic");
+            run("deminout");
+            run("opt");
+            run("opt_clean");
+            run("peepopt");
+            run("techmap");
+            run("muxcover -mux8 -mux4");
+            run("abc -luts 1,2,2,4");
+            run("opt");
+            run("check");
+        }
+
+        if (check_label("map")) {
+            run("techmap -map +/quicklogic/cells_map.v");
+            run("opt_clean");
+            run("check");
+            run("autoname");
+        }
+
+        if (check_label("iomap")) {
+            run("clkbufmap -buf $_BUF_ Y:A -inpad ckpad Q:P");
```

Comment:
With my recent `clkbufmap` improvements in #2230, the `-buf $_BUF_ Y:A` part is no longer necessary and can be dropped

Reply:
rebased and removed the `-buf` switch